packages/common/src/cis2/util.ts

```diff
@@ -201,6 +220,15 @@ function serializeCIS2TokenId(tokenId: HexString): Buffer {
     return packBufferWithWord8Length(serialized);
 }
 
+function deserializeCIS2TokenId(buffer: Buffer): CIS2.TokenId {
+    if (buffer.length > TOKEN_ID_MAX_LENGTH) {
+        throw Error(
+            `Token ID exceeds maximum size of ${TOKEN_ID_MAX_LENGTH} bytes`
+        );
+    }
+    return buffer.toString('hex');
```

Comment:
The first byte of the buffer will be the length of the ID, so you need to first read the first u8 and then read the rest. Also, it would make sense to either throw an error if the length of the remaining buffer exceeds the length read from the first byte (or return the end index as done with `uleb128DecodeWithIndex`?)

Reply:
My mistake, the spec says only the bytes of the token ID are included (which I misread to be including the length).. I think it looks all good now, sorry for the confusion.